src/TSTransformer/nodes/class/transformDecorators.ts

```diff
@@ -0,0 +1,289 @@
+import luau from "@roblox-ts/luau-ast";
+import { assert } from "Shared/util/assert";
+import { TransformState } from "TSTransformer";
+import { transformExpression } from "TSTransformer/nodes/expressions/transformExpression";
+import { transformObjectKey } from "TSTransformer/nodes/transformObjectKey";
+import ts from "typescript";
+
```

Comment:
This file could use a wrapper function, since a lot of the code is duplicated between the 3 decorator types.